Infinispan client deployment tests fail unless `test-containers` and `start-containers` is set

### Describe the bug

Building the Infinispan client deployment module will fail unless `test-containers` and `start-containers` is set.

### Expected behavior

It is expected that container-dependent tests would be skipped if `test-containers` is not set.

### Actual behavior

```
[ERROR] io.quarkus.infinispan.test.MultipleNamedInfinispanClientsTest.testNamedDevServices -- Time elapsed: 0.005 s <<< ERROR!
jakarta.enterprise.inject.CreationException: Null contextual instance was produced by a normal scoped synthetic bean: SYNTHETIC bean [class=org.infinispan.client.hotrod.RemoteCacheManager, id=iuJWECJKRxD6SGBV0Bxy9oFLf-0]
    at org.infinispan.client.hotrod.RemoteCacheManager_iuJWECJKRxD6SGBV0Bxy9oFLf-0_Synthetic_Bean.doCreate(Unknown Source)
    at org.infinispan.client.hotrod.RemoteCacheManager_iuJWECJKRxD6SGBV0Bxy9oFLf-0_Synthetic_Bean.create(Unknown Source)
    at org.infinispan.client.hotrod.RemoteCacheManager_iuJWECJKRxD6SGBV0Bxy9oFLf-0_Synthetic_Bean.create(Unknown Source)
    at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:118)
    at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
    at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
    at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.c3(Unknown Source)
    at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
    at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
    at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
    at org.infinispan.client.hotrod.RemoteCacheManager_iuJWECJKRxD6SGBV0Bxy9oFLf-0_Synthetic_ClientProxy.arc$delegate(Unknown Source)
    at org.infinispan.client.hotrod.RemoteCacheManager_iuJWECJKRxD6SGBV0Bxy9oFLf-0_Synthetic_ClientProxy.getConfiguration(Unknown Source)
    at io.quarkus.infinispan.test.MultipleNamedInfinispanClientsTest.testNamedDevServices(MultipleNamedInfinispanClientsTest.java:48)
    at java.base/java.lang.reflect.Method.invoke(Method.java:568)
    at io.quarkus.test.QuarkusUnitTest.runExtensionMethod(QuarkusUnitTest.java:500)
    at io.quarkus.test.QuarkusUnitTest.interceptTestMethod(QuarkusUnitTest.java:414)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```
and other exceptions.

### How to Reproduce?

Build Quarkus once (e.g. `mvn clean && mvn install -DskipTests`) and then run `mvn install` in the `extensions/infinispan-client/deployment` directory.

### Output of `uname -a` or `ver`

Darwin xxxx 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:28:45 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6020 arm64

### Output of `java -version`

openjdk version "17.0.9" 2023-10-17 OpenJDK Runtime Environment Temurin-17.0.9+9 (build 17.0.9+9) OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (build 17.0.9+9, mixed mode)

### Quarkus version or git rev

0d90b7c0b82262ed38a00ac8b34385f19c88dd9a

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.5 (57804ffe001d7215b5e7bcb531cf83df38f93546) Maven home: /Users/david/local/apache-maven Java version: 17.0.9, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home Default locale: en_US, platform encoding: UTF-8 OS name: "mac os x", version: "14.1.2", arch: "aarch64", family: "mac"

### Additional information

_No response_

--- a/extensions/infinispan-client/deployment/pom.xml
+++ b/extensions/infinispan-client/deployment/pom.xml
@@ -110,9 +110,39 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
     <profiles>
+        <profile>
+            <id>test-infinispan</id>
+            <activation>
+                <property>
+                    <name>test-containers</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>false</skip>
+                        </configuration>
+                    </plugin>
+                    <plugin>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                        <configuration>
+                            <skip>false</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
         <!-- Skip the tests on Windows -->
         <profile>
             <id>skip-tests-windows</id>

